src/views/routes/form/RouteForm.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { FC, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom-v5-compat';
+
+import { RouteModel } from '@kubevirt-ui/kubevirt-api/console';
+import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  ActionGroup,
+  Alert,
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  Checkbox,
+  Form,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  PageSection,
+  PageSectionVariants,
+  Text,
+  TextInput,
+  TextVariants,
+  Title,
+} from '@patternfly/react-core';
+import { ALL_NAMESPACES_KEY, DEFAULT_NAMESPACE } from '@utils/constants';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { RouteKind } from '@utils/types';
+import { isEmpty, resourcePathFromModel } from '@utils/utils';
+import { generateName } from '@utils/utils/utils';
+
+import ServiceSelector from './ServiceSelector';
+import TLSTermination from './TLSTermination';
+
+type RouteFormProps = {};
+
+const RouteForm: FC<RouteFormProps> = () => {
+  const { t } = useNetworkingTranslation();
+  const navigate = useNavigate();
+  const [apiError, setError] = useState<Error>(null);
+  const [activeNamespace] = useActiveNamespace();
+  const namespace = ALL_NAMESPACES_KEY === activeNamespace ? DEFAULT_NAMESPACE : activeNamespace;
```

Comment:
I recommend adding a function in src/utils like the one we have in the kubevirt-plugin for this.

```
export const getValidNamespace = (activeNamespace: string) =>
  activeNamespace === ALL_NAMESPACES_SESSION_KEY ? DEFAULT_NAMESPACE : activeNamespace;
```